bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/SecureSocketRequest.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.security.api;
+
+import java.util.Set;
+
+import javax.net.ssl.SSLContext;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Implement this to request a secure socket implementation (SSLEngine or others) and get notified of the response.
+ * Use {@link SecureSocketServers#requestSecureSocketImplementation(SecureSocketRequest)} to issue a request.
+ *
+ * {@link SecureSocketServers} will also notify you about changes of the requested secure socket implementation
+ * during the life-time of the service.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface SecureSocketRequest {
+    /**
+     * @return Return mandatory capabilities. They are required for a requested secure socket implementation.
+     *         If you need a SSLEngine or you need DTLS support, you would specify this here. Please make yourself
+     *         familiar with the java platform and the ESH backend service limitations. You cannot for example
+     *         request a SSLEngine with DTLS support in Java 8, this is only an option on Java 9+.
+     */
+    Set<SecureSocketCapability> mandatoryCapabilities();
+
+    /**
+     * @return Return optional capabilities. They are considered if possible to match the best secure socket
+     *         implementation. If you prefer a CA-signed certificate over a self-signed one for example, you would
+     *         express this here.
+     */
+    Set<SecureSocketCapability> optionalCapabilities();
+
+    /**
+     * Returns a potentially new secure socket implementation instance that fulfills the requested mandatory
```

Comment:
If you say that this method should `return` something, why is it `void`? :)

Since you use this `SecureSocketRequest` class as a callback object, maybe you should make this clear that this methods sets the socket implementation on this callback object but does not `return` it.